ddsp/training/docker/user_interaction.py

```diff
@@ -1,84 +1,235 @@
+script that can be called directly.
+================================================================================
+
+ddsp_ai_platform \
+--data_path=gs://ddsp_training/data \
+--save_dir=gs://ddsp_training/model \
+--batch_size=128\
+--learning_rate=0.001
+
+
+"""
 
 import os
+import datetime
+import subprocess
+
+from absl import app
+from absl import flags
+from google.cloud import storage
+
+FLAGS = flags.FLAGS
+
+# Program flags.
+flags.DEFINE_string('data_path', '',
+                    'Path where the dataset for training is saved')
+flags.DEFINE_string('save_dir', '',
+                    'Path where checkpoints and summary events will be saved '
+                    'during training and evaluation.')
+flags.DEFINE_string('restore_dir', '',
+                    'Path from which checkpoints will be restored before '
+                    'training. Can be different than the save_dir.')
+flags.DEFINE_string('config_path', '',
+                    'Configuration file for AI Platform training')
+flags.DEFINE_string('project_id', '', 'GCP Project ID')
+flags.DEFINE_string('region', '', 'GCP region for running the training')
+flags.DEFINE_string('batch_size', '', 'Batch size')
+flags.DEFINE_string('learning_rate', '', 'Learning rate')
+flags.DEFINE_string('num_steps', '', 'Number of training steps')
+flags.DEFINE_string('steps_per_summary', '', 'Steps per summary')
+flags.DEFINE_string('steps_per_save', '', 'Steps per save')
+flags.DEFINE_string('early_stop_loss_value', '',
+                   'Early stopping. When the total_loss reaches below this '
+                   'value training stops.')
+
+def prompt(default, message, skip=False, arg='', local=False):
+  input_prompt = 'Insert '
+  if default == 'path':
+    input_prompt += 'the path to where the '
+    input_prompt += message
+    if not local:
+      input_prompt += '. This must be a GCS Bucket'
+    if skip:
+      input_prompt += '. Skip for the default one'
+    input_prompt += ':\n'
+  if default == 'value':
+    input_prompt += message
+    if skip:
+      input_prompt += ' or skip for the default value (' + arg + ')'
+    input_prompt += ':'
+  return input_prompt
+
+def check_bucket(bucket_name, message):
+  while not bucket_name.startswith('gs://'):
+    print('Path not a bucket.')
+    bucket_name = input(message)
+  while True:
+    try:
+      name = bucket_name[5:]
+      if '/' in name:
+        name = name.split('/')[0]
+      storage.Client().get_bucket(name)
+      break
+    except:
+      print('Not a valid bucket.')
+      bucket_name = input(message)
+  return bucket_name
 
 def get_input():
   """Gathers input from user"""
   # Determining path for data retrieval
   input_dict = dict()
-  data_path = input('Insert a path for data retrieving. '
-                      'This must be a GCS Bucket:\n')
+  if FLAGS.data_path:
+    data_path = FLAGS.data_path
+  else:
+    data_path = input(prompt('path', 'dataset for training is stored'))
```

Comment:
Also, see above, but prompt messages should be full statements at the arguments level so they are readable.

Example
```
data_path = prompt_path('Path to training dataset directory', required=True)
```
-> "Path to training dataset directory (Required):"